server/memphis_handlers_user_mgmt.go

```diff
@@ -188,18 +188,22 @@ func removeTenantResources(tenantName string) error {
 		return err
 	}
 
-	for _, station := range stations {
-		stationName := strings.ToLower(station.Name)
-		sName, err := StationNameFromStr(stationName)
-		if err != nil {
-			return err
-		}
-
-		err = serv.RemoveStream(tenantName, sName.Intern())
+	err = serv.memphisPurgeResourcesAccount(tenantName)
+	if err != nil {
 		if err != nil && !IsNatsErr(err, JSStreamNotFoundErr) {
+			serv.Errorf("Failed deleting account %s - %s", tenantName, err.Error())
 			return err
 		}
+	}
 
+	for _, user := range users {
```

Comment:
you only need to reload once not once per user

Reply:
hence you don't need to call GetApplicationUsersByTenantName and this db function is unused beside it